kernel/src/hil/i2c.rs

```diff
@@ -27,19 +28,27 @@ pub enum Error {
 
     /// No error occured and the command completed successfully.
     CommandComplete,
```

Comment:
I agree with your ML comment that having `CommandComplete` here no longer makes sense -- it might be better to remove this enum variant and instead have the `command_complete()` callback pass a `Result<(), Error>`

Reply:
This make much more sense. On the other hand, as you wrote in the mailing list, if some HILs return different, that might be a problem with converting errors.